src/high-level/tests/KinDynComputationsUnitTest.cpp

```diff
@@ -260,6 +278,122 @@ void testInverseDynamics(KinDynComputations & dynComp)
+        // on the getFrameVelocityRepresentation used:
+        //
+        // |`FrameVelocityRepresentation`     |       C      |
+        // |:--------------------------------:|:------------:|
+        // | `MIXED_REPRESENTATION` (default) | \f$ L[A] \f$ |
+        // | `BODY_FIXED_REPRESENTATION`      | \f$   L  \f$ |
+        // | `INERTIAL_FIXED_REPRESENTATION`  | \f$   A \f$  |
+
+        // The inertial term is a complicated to compute in mixed representation, so for now we
+        // test only BODY_FIXED_REPRESENTATION or INERTIAL_FIXED_REPRESENTATION
+        if (dynComp.getFrameVelocityRepresentation() == MIXED_REPRESENTATION) {
+            return;
+        }
+
+        iDynTree::LinkWrenches inertialWrenches(dynComp.model());
+        iDynTree::LinkWrenches externalWrenches(dynComp.model());
+        iDynTree::LinkWrenches internalWrenches(dynComp.model());
+
+        iDynTree::VectorDynSize dummyJointPos, dummyJointVel;
+        iDynTree::Vector3 gravityInAbsoluteFrame;
+        dummyJointPos.resize(dynComp.model().getNrOfPosCoords());
+        dummyJointVel.resize(dynComp.model().getNrOfDOFs());
+        dynComp.getRobotState(dummyJointPos, dummyJointVel, gravityInAbsoluteFrame);
+
+        LinkIndex baseLnkIdx = dynComp.getRobotModel().getLinkIndex(dynComp.getFloatingBase());
+
+        for(LinkIndex lnkIdx = 0; lnkIdx < dynComp.getNrOfLinks(); lnkIdx++) {
+            // Compute inertialWrenches
+            // Inertia is always expressed in L, let's convert it in A if we are in INERTIAL_FIXED_REPRESENTATION
+            iDynTree::SpatialInertia rawInertia = dynComp.getRobotModel().getLink(lnkIdx)->getInertia();
+            iDynTree::SpatialInertia I;
+            if (dynComp.getFrameVelocityRepresentation() == INERTIAL_FIXED_REPRESENTATION) {
+                I = dynComp.getWorldTransform(lnkIdx)*rawInertia;
+            } else {
+                // BODY_FIXED_REPRESENTATION
+                I = rawInertia;
+            }
+
+            // Not really efficient, but ok as we are in a test
+            iDynTree::SpatialAcc     a = Vector6ToSpatialAcc(dynComp.getFrameAcc(lnkIdx, baseAcc,shapeAccs));
+            iDynTree::Twist          v = dynComp.getFrameVel(lnkIdx);
+            // Add "gravitational" force
+            iDynTree::Wrench gravitationalWrench;
+            if (dynComp.getFrameVelocityRepresentation() == INERTIAL_FIXED_REPRESENTATION) {
+                gravitationalWrench = (I*Vector3ToSpatialAcc(gravityInAbsoluteFrame));
+            } else {
+                // BODY_FIXED_REPRESENTATION
+                gravitationalWrench = (I*(dynComp.getWorldTransform(lnkIdx).inverse()*Vector3ToSpatialAcc(gravityInAbsoluteFrame)));
+            }
+            inertialWrenches(lnkIdx) = I*a + v*(I*v) - gravitationalWrench;
+
+
+
+            inertialWrenches(lnkIdx) = inertialWrenches(lnkIdx) ;
```

Comment:
Maybe I am missing something, but why is this line needed?

Reply:
It is a left-over, we can remove it, thanks for noticing.